sharktank/sharktank/ops/sharded_impls.py

```diff
@@ -30,6 +30,110 @@
 from .shape import broadcast_dims, broadcast_dim, unbroadcast_dim
 from ..utils import longest_equal_range
 
+def tensor_w_shards_moved(tensor: ShardedTensor, new_devices: Tuple[int]) -> ShardedTensor:
+    """
+    Create a copy of the passed in tensor, but with shards now placed on the new devices.
+    """
+    # TODO: What does transfrom_globals need from this function?
+
+    new_shards = tuple(
+        (
+            transfer_to_logical_device(shard, new_devices[j])
+            if new_devices[j] != tensor.devices[j]
+            else barrier_on_logical_device(shard, new_devices[j])
+        )
+        for j, shard in enumerate(tensor.shards)
+    )
+
+    if isinstance(tensor, SplitPrimitiveTensor ):
+        return SplitPrimitiveTensor(
```

Comment:
One thing we may want to do is create a `clone` operation that takes all of the normal constructor args then overrides with the changed ones. This way you could do
```
tensor.clone(ts=new_shards, pinned=False, devices=new_devices)
```
and be guaranteed it just makes a new cloned version with the right devices and pinning